mt-kahypar/partition/refinement/gains/gain_computation_base.h

```diff
@@ -155,11 +155,8 @@ class GainComputationBase {
   }
 
   void changeNumberOfBlocks(const PartitionID new_k) {
-    for ( auto& tmp_score : _tmp_scores ) {
-      if ( static_cast<size_t>(new_k) > tmp_score.size() ) {
-        tmp_score = RatingMap(new_k);
-      }
-    }
+    TmpScores new_tmp_scores(new_k);
```

Comment:
The previous function only resizes a rating map if its size is less than k. In the new implementation, we deallocate all rating maps and replace them with a new ones in all cases. Use here a similar solution as in the problem construction class.